### ISSUE TITLE
Question about disabling the causal mask

### ISSUE THREAD

**Issue by volcverse** [Dao-AILab/flash-attention#1304]
Hello, thanks for the great work!

I'm currently working on something with customized attention_mask. But I didn't find any way to do so in flash attention, except changing the causal=True to causal=False.

So, does the FlashAttention now only support the triangle mask (causal=True)? When causal=False, is the attention mask an all-one matrix? So each query attends to all the keys right?

Any response will be greatly appreciated!
